README.md

```diff
@@ -78,6 +78,117 @@ createAction('ADD_TODO')('Use Redux');
 
 `metaCreator` is an optional function that creates metadata for the payload. It receives the same arguments as the payload creator, but its result becomes the meta field of the resulting action. If `metaCreator` is undefined or not a function, the meta field is omitted.
 
+### `createActions(?actionMap, ?arrayDefTypes, ?optionMap)`
+
+Create multiple actions and return an object mapping from the type to the actionCreator. This eliminates some of the repetitiveness when creating many actions.
+
+If `actionMap` is provided as first argument, an action will be created for each item in the map. The key will be used for the type. The value will be treated as follows:
+
+ - if value is a function then it will be assumed to be the payloadCreator function
+ - if value is an object then it will accept as its keys, `payload` and `meta`, both of which are functions that will serve as the payloadCreator and/or metaCreator. If payloadCreator is missing the default payloadCreator is used. If metaCreator is missing then it will not use one.
+ - otherwise create a default actionCreator for the key type
+
+If an array of string is provided as `arrayDefTypes` it will be used to create default actionCreators using the string for the action type.
+
+If an `optionMap` object is supplied as last argument then it will adjust the options createActions uses in operation:
+
+ - `prefix` - string, prefix all action types with this prefix, default ''
+
+#### Example: simple default actions
+
+```js
+import { createActions } from 'redux-actions';
+const a = createActions(['ONE', 'TWO', 'THREE']);
+
+// is equivalent to
+const a = {
+  ONE: createAction('ONE'),
+  TWO: createAction('TWO'),
+  THREE: createAction('THREE')
+;}
+```
+
+#### Example: simple default actions with prefix
+
+```js
+import { createActions } from 'redux-actions';
+const a = createActions(['ONE', 'TWO', 'THREE'], { prefix: 'ns/');
+
+// is equivalent to
+const a = {
+  ONE: createAction('ns/ONE'),
+  TWO: createAction('ns/TWO'),
+  THREE: createAction('ns/THREE')
```

Comment:
1. I leave the case up to the user. If you want to use caps (snake case) or
camel case you can do so, but whatever is passed is by default the keys to
lookup the action by. I personally use camel case myself.

As for the prefix, the thinking is that it is nice to namespace the action
types so you don't have to worry about collisions. All types in a file get
that namespace prefix. I didn't find it necessary to add the prefix to the
keys though, we could but it seems nicer to skip that and just use what was
passed. If we were to add prefix to the keys too then you might have to use
brackets to reference the action creator which is not nice. So the prefix
seems best to only add to the type and that's where it is needed, it really
isn't needed for the keys since they are already unique in the file.

I guess if that is too confusing I could rename it prefixType or we could
have separate options, one for prefixing both and one for only prefixing
types. Options might be prefixType (only type) and prefix (do both). I just
didn't think most people would want both, code seems better doing just
keys.


2. I don't have an expectation for camel casing, use what you like.

If others request it we could have additional options which transform the
strings for the action types'. We could have an option that makes them use
upper snake casing or one that formats as camel case, etc. I left the
options open that we could support that but I didn't want to add that until
enough people wanted it.


If you have other ideas on how to improve the API or things that are
confusing let me know. Could they be solved with clearer readme?

Thanks,

Jeff


On Saturday, July 16, 2016, Victor Alvarez <notifications@github.com> wrote:

> In README.md
> <https://github.com/acdlite/redux-actions/pull/104#discussion_r71063375>:
>
> > +  TWO: createAction('TWO'),
> > +  THREE: createAction('THREE')
> > +;}
> > +```
> > +
> > +#### Example: simple default actions with prefix
> > +
> > +```js
> > +import { createActions } from 'redux-actions';
> > +const a = createActions(['ONE', 'TWO', 'THREE'], { prefix: 'ns/');
> > +
> > +// is equivalent to
> > +const a = {
> > +  ONE: createAction('ns/ONE'),
> > +  TWO: createAction('ns/TWO'),
> > +  THREE: createAction('ns/THREE')
>
> What is the expectation around the keys
>
>    1. having the full action name (including the prefix when it exists)
>    2. being camel-cased
>
> My thoughts:
>
>    1. is that it seems like surprising behavior that the prefix be
>    omitted 2.we're returning functions (action creators), and having the name
>    by the action type is potentially confusing
>
> Thoughts?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/acdlite/redux-actions/pull/104/files/628cf27017ee8dac86e5f1b08606b826e753e056#r71063375>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAWOebMQo6eBwlijvsZJIL31A1BA42gks5qWJxhgaJpZM4JIZ-x>
> .
>


-- 
Jeff Barczewski
Founder of CodeWinds
http://codewinds.com/
Online developer training